Flaky CI failures in proxy-based replication tests

Several recent `kvrocks.yaml` CI runs failed in replication tests that simulate broken or slow replication links.

The most frequent failure is `TestSlaveLostMaster`:
```
--- FAIL: TestSlaveLostMaster
    client.go:147: forward tcp stream failed, err: read tcp 127.0.0.1:...: read: connection reset by peer
FAIL github.com/apache/kvrocks/tests/gocase/integration/replication
```

Examples:
- https://github.com/apache/kvrocks/actions/runs/27181757902
- https://github.com/apache/kvrocks/actions/runs/27136241761
- https://github.com/apache/kvrocks/actions/runs/27135463274
- https://github.com/apache/kvrocks/actions/runs/26640257024
- https://github.com/apache/kvrocks/actions/runs/26447014065

This does not seem related to the PR contents. It has appeared across unrelated stream/cf/hash/dependency changes and across different CI matrix entries, especially macOS arm64, but not only there.

The failure appears to come from `util.SimpleTCPProxy` calling `t.Fatalf` from the proxy forwarding goroutine when the TCP connection is reset. In `TestSlaveLostMaster`, the test intentionally breaks the proxy connection to simulate a lost master, so a connection reset during that phase may be expected and should probably not fail the test directly.

Proxy-based tests worth auditing:
- `TestSlaveLostMaster`
- `TestSlowConsumerBug`
- `TestSlowConsumerBlocksIndefinitely`

### Execute the necessary steps
- Investigate the reasons for failure
- Confirm the purpose of the test cases
- Propose solutions (communication is very important)
- Complete the code implementation

--- a/tests/gocase/util/client.go
+++ b/tests/gocase/util/client.go
@@ -28,6 +28,7 @@ import (
 	"regexp"
 	"strings"
 	"sync/atomic"
+	"syscall"
 	"testing"
 	"time"
 
@@ -104,14 +105,17 @@ func SimpleTCPProxy(ctx context.Context, t testing.TB, to string, slowdown bool)
 					}
 					n, err := src.Read(buffer)
 					if err != nil {
-						if errors.Is(err, io.EOF) {
+						// ECONNRESET can happen instead of io.EOF when the peer closes
+						// the connection abortively (e.g. with unread data still
+						// buffered), which is expected during intentional teardown.
+						if errors.Is(err, io.EOF) || errors.Is(err, syscall.ECONNRESET) {
 							break COPY_LOOP
 						}
 						return err
 					}
 					_, err = dest.Write(buffer[:n])
 					if err != nil {
-						if errors.Is(err, io.EOF) {
+						if errors.Is(err, io.EOF) || errors.Is(err, syscall.ECONNRESET) {
 							break COPY_LOOP
 						}
 						return err

